### PR TITLE
Add support for internal load balancing

### LINKED ISSUE #116: Add internal load balancer support to CPI

### PR DESCRIPTION
**Pull request by danawillow**
This enables VMs to be placed in Region Backend Services by allowing the BackendService field to point to a single string (in which case, it represents a regular Backend Service) or a map of strings with name and region fields (for a Region Backend Service).

Fixes #116

### CODE REVIEW

File: src/bosh-google-cpi/action/create_vm.go [hunk 1]
```diff
@@ -113,6 +118,17 @@ func (cv CreateVM) Run(agentID string, stemcellCID StemcellCID, cloudProps VMClo
 		return "", bosherr.WrapError(err, "Creating VM")
 	}
 
+	var bs BackendService
+	if bsString, ok := cloudProps.BackendService.(string); ok {
+		bs = BackendService{bsString, ""}
```
Comment (evandbrown) [inline src/bosh-google-cpi/action/create_vm.go]: Let's name the fields when initializing the struct to keep it consistent and easier to read (i.e., BackendService{name: ..., scheme: ...,}

File: src/bosh-google-cpi/integration/vm_test.go [hunk 2]
```diff
@@ -637,7 +637,7 @@ var _ = Describe("VM", func() {
 				{
 				  "machine_type": "n1-standard-1",
 				  "zone": "%v",
-				  "backend_service": "%v"
+				  "backend_service": {"name": "%v", "scheme": "external"}
```
Comment (evandbrown) [inline src/bosh-google-cpi/integration/vm_test.go]: Can we make this a separate test, so we preserve the original test for backwards compat?

**Comment by danawillow** [comment]
I think you're looking at an outdated version- both of those should already have been fixed.

**Comment by evandbrown** [comment]
Oh that is weird. The GitHub review pane shows the older version. Wild. LGTM!

**Comment by evandbrown** [comment]
@danawillow: 1 integration [test failed](https://107.178.243.59/teams/main/pipelines/develop-bosh-google-cpi-release/jobs/run-int/builds/84):

```
STEP: creating a VM
• Failure [23.594 seconds]
VM
/tmp/build/e797b25d/bosh-cpi-src/src/bosh-google-cpi/integration/vm_test.go:726
  executes the VM lifecycle with a region backend service [It]
  /tmp/build/e797b25d/bosh-cpi-src/src/bosh-google-cpi/integration/vm_test.go:724

  Expected
      <*dispatcher.ResponseError | 0xc8203556e0>: {
          Type: "Bosh::Clouds::VMCreationFailed",
          Message: "VM failed to create: Backend Service \"develop-ci\" does not exist: <nil cause>",
          CanRetry: true,
      }
  to be nil
```

**Pull request merged**